aiida/manage/database/delete/nodes.py

```diff
@@ -86,24 +74,23 @@ def delete_nodes(
             echo.echo('Nothing to delete')
         return
 
+    follow_forwards = []
     follow_backwards = []
-    follow_backwards.append(LinkType.CREATE.value)
-    follow_backwards.append(LinkType.RETURN.value)
-    follow_backwards.append(LinkType.CALL_CALC.value)
-    follow_backwards.append(LinkType.CALL_WORK.value)
 
-    follow_forwards = []
-    follow_forwards.append(LinkType.INPUT_CALC.value)
-    follow_forwards.append(LinkType.INPUT_WORK.value)
+    # Create the dictionary with graph traversal rules to be used in determing complete node set to be exported
+    for name, rule in GraphTraversalRules.DELETE.value.items():
+
+        # Check that rules that are not toggleable are not specified in the keyword arguments
+        if not rule.toggleable and name in kwargs:
+            raise exceptions.ExportValidationError('traversal rule {} is not toggleable'.format(name))
 
-    if create_forward:
-        follow_forwards.append(LinkType.CREATE.value)
+        follow = kwargs.pop(name, rule.default)
 
-    if call_calc_forward:
-        follow_forwards.append(LinkType.CALL_CALC.value)
+        if follow and rule.direction == 'forward':
+            follow_forwards.append(rule.link_type.value)
 
-    if call_work_forward:
-        follow_forwards.append(LinkType.CALL_WORK.value)
+        elif follow and rule.direction == 'backward':
+            follow_backwards.append(rule.link_type.value)
```

Comment:
I'm going to approve, but do we want to put here (or somewhere else?) something to check the `else` clause to verify we didn't mistype some of the values, being neither `forward` nor `backward`? E.g. raising an InternalError

Reply:
Done